internal/provider/kubernetes/controller.go

```diff
@@ -329,15 +339,17 @@ func (r *gatewayAPIReconciler) Reconcile(ctx context.Context, request reconcile.
 		service := new(corev1.Service)
 		err := r.client.Get(ctx, serviceNamespaceName, service)
 		if err != nil {
-			r.log.Error(err, "unable to find associated Services")
 			if kerrors.IsNotFound(err) {
-				return reconcile.Result{}, nil
+				r.log.Info("failed to find Service", "namespace", serviceNamespaceName.Namespace,
+					"name", serviceNamespaceName.Name, "error", err)
+			} else {
+				return reconcile.Result{}, err
```

Comment:
shouldnt this `return` be removed as well ?

Reply:
Good catch. Yes, commit 365610b  fixes this.